collab-service/src/message_handler/room/connect_message_builder.ts

```diff
@@ -8,22 +9,37 @@ export const OPCODE_QUESTION_RCV = 9;
 export const OPCODE_SAVE_CODE_SEND = 10;
 export const OPCODE_SAVE_CODE_ACK = 11;
 
-function createConnectionAlertMessage(username: string, opcode: number): Uint8Array {
+function createConnectionAlertPackage(content: string, opcode: number): Uint8Array {
   const encoder = encoding.createEncoder();
   encoding.writeUint8(encoder, opcode);
-  encoding.writeVarString(encoder, username);
+  encoding.writeVarString(encoder, content);
   return encoding.toUint8Array(encoder);
 }
 
-function createConnectedMessage(username: string): Uint8Array {
-  return createConnectionAlertMessage(username, OPCODE_USER_JOIN);
+function createConnectedPackage(username: string): Uint8Array {
+  return createConnectionAlertPackage(username, OPCODE_USER_JOIN);
 }
 
-function createDisconnectedMessage(username: string): Uint8Array {
-  return createConnectionAlertMessage(username, OPCODE_USER_LEAVE);
+function createDisconnectedPackage(username: string): Uint8Array {
+  return createConnectionAlertPackage(username, OPCODE_USER_LEAVE);
+}
+
+function createQuestionRcvPackage(question: string): Uint8Array {
+  return createConnectionAlertPackage(question, OPCODE_QUESTION_RCV);
+}
+
+/*
+ * Extracts opcode of given data package
+ * @param data
+ */
+function readConnectionOpCode(data: Uint8Array): number {
+  const decoder = decoding.createDecoder(data);
+  return decoding.readUint8(decoder);
```

Comment:
I believe it will work if you simply read index 0 of the array, no need to create an entire decoder for this.